src/components/Button/ButtonProps.types.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  ColorValue,
+  GestureResponderEvent,
+  StyleProp,
+  TextStyle,
+  TouchableOpacityProps,
+} from 'react-native';
+
+export type ButtonIconProps = {
+  icon?: string;
```

Comment:
If we had used CustomIcon here then this would be of type CustomIconNames, making typescript better

Reply:
Fixed it now.